examples/reading_and_writing.py

```diff
@@ -0,0 +1,54 @@
+"""
```

Comment:
can we add a test that runs this example? I think we could even just `pytest.monkeypatch` the `print` function and check the output directly

Reply:
I definitely think that's a good idea, but it should probably be in a new feature branch, and won't effect release code.

Reply:
Is example code usually included in the unit tests? I can't recall ever seeing this, but I suppose it makes sense. Personally, I always thought of sample code more as part of the documentation.

Reply:
it definitely *is* just the documentation. but I'd still like to know that the example works; it'd be embarrassing to have someone copy the example and have it not work.

Reply:
@CFlaniganMide agreed, this can just go in a feature branch elsewhere